### PR TITLE
Try-with-resource clean-up in JDT core

### PR DESCRIPTION
**Pull request by vogella**
Using the JDT UI try-with-resource clean-up on core

## Author checklist

- [ ] I have thoroughly tested my changes
- [ ] The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
- [ ] I have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)

### CODE REVIEW

**Comment by vogella** [comment]
@jukzi as you are doing this work in other repos, can you review?

File: org.eclipse.jdt.apt.pluggable.tests/src/org/eclipse/jdt/apt/pluggable/tests/processors/buildertester/Bug468893Processor.java [hunk 1]
```diff
@@ -43,7 +43,7 @@ public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment
 			try {
 				FileObject resource = processingEnv.getFiler().createSourceFile("generated.TypeIndex");
 				BufferedWriter w = new BufferedWriter(resource.openWriter());
```
Comment (jukzi) [inline org.eclipse.jdt.apt.pluggable.tests/src/org/eclipse/jdt/apt/pluggable/tests/processors/buildertester/Bug468893Processor.java:45]: better put into try argument   -looks nicer

File: org.eclipse.jdt.apt.tests/src-annotations/org/eclipse/jdt/apt/tests/annotations/readAnnotationType/ReadAnnotationTypeProcessor.java [hunk 2]
```diff
@@ -32,7 +32,7 @@ public ReadAnnotationTypeProcessor(Set<AnnotationTypeDeclaration> declarationTyp
 
     public void process() {
         PrintWriter writer = null;
-        try
+        try (writer)
```
Comment (jukzi) [inline org.eclipse.jdt.apt.tests/src-annotations/org/eclipse/jdt/apt/tests/annotations/readAnnotationType/ReadAnnotationTypeProcessor.java:35]: wrtier is `null`, this seems to be wrong

**Review by jukzi — CHANGES_REQUESTED** [review]
there are many cases where it would look better to have the constructor in the argument, i have not marked all

File: org.eclipse.jdt.apt.tests/src/org/eclipse/jdt/apt/tests/PerfTests.java [hunk 3]
```diff
@@ -70,12 +70,9 @@ public void setUp() throws Exception
 		
 		InputStream in = new FileInputStream(tempPath);
 		ZipInputStream zipIn = new ZipInputStream(in);
```
[the remark this answers is not in the document]
Comment (vogella) [reply org.eclipse.jdt.apt.tests/src/org/eclipse/jdt/apt/tests/PerfTests.java:72]: Done

File: org.eclipse.jdt.compiler.apt.tests/processors8/org/eclipse/jdt/compiler/apt/tests/processors/elements/Java8ElementProcessor.java [hunk 4]
```diff
@@ -1095,7 +1095,7 @@ private void createPackageBinary() throws IOException {
 		String path = packageName.replace('.', '/');
 		ClassLoader loader = getClass().getClassLoader();
 		InputStream in = loader.getResourceAsStream(path + "/package-info.class");
```
[the remark this answers is not in the document]
Comment (vogella) [reply org.eclipse.jdt.compiler.apt.tests/processors8/org/eclipse/jdt/compiler/apt/tests/processors/elements/Java8ElementProcessor.java:1097]: Done

**Pull request merged**